pkg/prototype/prototype.go

```diff
@@ -189,6 +200,25 @@ type ParamSchema struct {
 	Type        ParamType `json:"type"`
 }
 
+// IsRequired returns if this parameter is required. Parameters are required
+// if the default value is nil.
+func (ps *ParamSchema) IsRequired() bool {
+	return ps.Default == nil
+}
+
+func (ps *ParamSchema) QuotedValue(value string) (string, error) {
```

Comment:
Is it ever valid to set a parameter to an empty string? If so, we'll need a way to distinguish between that case and the case where no value was given and default should be used.

Reply:
related: https://github.com/ksonnet/ksonnet/issues/273